nixos/tests/systemd-initrd-networkd.nix

```diff
@@ -0,0 +1,45 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }: {
+  name = "systemd-initrd-network";
+  meta.maintainers = [ lib.maintainers.elvishjerricco ];
+
+  nodes = {
+    basic = { ... }: {
+      boot.initrd.network.enable = true;
+
+      boot.initrd.systemd = {
+        enable = true;
+        network.networks."99-eth0" = {
+          matchConfig.Name = "eth0";
```

Comment:
Will we have predicitve interface naming in the initrd?

Reply:
This was removed in a later commit in the commit titled `systemd-initrd: Automatically configure networking.interfaces`. It should be handled exactly the same way as stage 2 does it. Is there more that needs to be done for predictable interface naming?

Reply:
Predictable names come from udev, so if udev is around to do it, probably not.